lib/src/entrypoint.dart

```diff
@@ -396,24 +277,12 @@ class Entrypoint {
       // SDK's.
       writeTextFile(p.join(_snapshotPath, 'sdk-version'), "${sdk.version}\n");
 
-      var packagesToLoad =
-          unionAll(executables.keys.map(packageGraph.transitiveDependencies))
-              .toSet();
-
-      // Try to avoid starting up an asset server to precompile packages if
-      // possible. This is faster and produces better error messages.
-      if (packagesToLoad
-          .every((package) => package.pubspec.transformers.isEmpty)) {
-        await _precompileExecutablesWithoutBarback(executables);
-      } else {
-        await _precompileExecutablesWithBarback(executables, packagesToLoad);
-      }
+      await _precompileExecutables(executables);
     });
   }
 
   //// Precompiles [executables] to snapshots from the filesystem.
-  Future _precompileExecutablesWithoutBarback(
-      Map<String, List<AssetId>> executables) {
+  Future _precompileExecutables(Map<String, List<AssetId>> executables) {
```

Comment:
For example, the package names in the `executables` map are totally redundant; this could just be `Map<String, List<String>>`. The only reason we represented executables as `AssetId`s before was for convenience when passing them to Barback.